_posts/2016-10-08-eidolon-deploy-process.markdown

```diff
@@ -0,0 +1,192 @@
+
+We also want to set the Info.plist's version to the one specified when we run the `fast lane` command.
+
+```rb
+build_number = Time.new.strftime("%Y.%m.%d")
+increment_build_number build_number: build_number
+
+increment_version_number version_number: version
+```
+
+Okay, now it's time to generate markdown release notes from the changelog. Our changelog is in the following format:
+
+```yaml
+upcoming:
+- Upcoming version bug fix.
+
+releases:
+- version: X.Y.Z
+  date: Month Day Year
+  notes:
+  - Previous version bug fix.
+```
+
+We want to grab the `upcoming` notes for the changelog, and then move them to the `releases` section. Let's generate the notes first:
+
+```rb
+changelog_filename = '../CHANGELOG.yml'
+changelog_yaml = YAML.load_file(changelog_filename)
+release_notes = changelog_yaml['upcoming'].map{ |note| note.prepend '- ' }.join("\n")
+```
+
+Updating the changelog is a little messy. I tried parsing the changelog as yaml, modifying it, and then writing it back as yaml, but kept running into trouble. Instead, I treat it as plain text. We open the changelog, split on `releases:`, prepend the existing releases with a the generated release notes, and write the changelog.
+
+```rb
+changelog_contents = File.read(changelog_filename)
+existing_releases = changelog_contents.split('releases:').last
+this_release = changelog_yaml['upcoming'].map{ |note| note.prepend '  ' }.join("\n")
+changelog_contents = <<-EOS
+upcoming:
+releases:
+- version: #{version}
+  date: #{Time.new.strftime("%B %d %Y")}
+  notes:
+#{this_release}
+#{existing_releases}
+EOS
+
+File.open(changelog_filename, 'w') { |file| file.puts changelog_contents }
+```
+
+At this point, we're ready to start the actual deploy process. First we need to download the provisioning profiles, which is only one step with Fastlane:
+
+```rb
+sigh
+```
+
+Next we build our app using `gym`. We need to use the legacy build API, I can't remember why.
+
+```rb
+gym(
+  scheme: "Kiosk",
+  export_method: 'enterprise',
+  use_legacy_build_api: true
+)
+```
+
+With our build finished, we upload to Hockey.
+
+```rb
+hockey(
+  api_token: hockey_api_token,
+  notes: release_notes
+)
+```
+
+Okay, our build is deployed. Time to let the team know there's a new version available:
+
+```rb
+slack(
+  message: "There is a new version of the Kiosk app available. Download it at http://artsy.net/kioskbeta",
+  success: true,        # optional, defaults to true
+  payload: {            # optional, lets you specify any number of your own Slack attachments
+    'Version' => version,
+    'What\'s new' => release_notes,
+  },
+  default_payloads: [],
+)
+```
+
+`default_payloads` needs to be empty I think, I can't remember why. Seems like "I can't remember why" is a common theme here...
+
+Before committing the changes we've made to the changelog and Info.plist files, we need to clean any build artefacts.
+
+```rb
+clean_build_artifacts
+```
```

Comment:
may be worth explaining what that does, I don't know what that does - I can guess but I'm willing to be I don't _really_ know